cinematch/views.py

```
@@ -0,0 +1,41 @@
+            # Find movies with similar embeddings using cosine distance
+            similar_movies = (
+                Movie.objects.filter(embedding__isnull=False)
+                .annotate(similarity_score=1 - CosineDistance("embedding", query_embedding))
+                .order_by("-similarity_score")
+                .prefetch_related("genre", "talent")
+            )
+
+            return similar_movies
+
+        except Exception as e:
+            raise ValidationError(f"Failed to generate recommendations: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)